src/main/kotlin/com/bugsnag/android/gradle/internal/UploadRequestClient.kt

```diff
@@ -0,0 +1,67 @@
+package com.bugsnag.android.gradle.internal
+
+import com.bugsnag.android.gradle.AndroidManifestInfo
+import org.gradle.api.Project
+import org.gradle.api.provider.Provider
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters
+import java.util.Objects
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.FutureTask
+
+sealed class UploadRequestClient : AutoCloseable {
+
+    private val requestMap = ConcurrentHashMap<String, FutureTask<String>>()
+
+    /**
+     * Executes request with automatic de-duplication of similar requests.
+     *
+     * If the version information and payload match another request that
+     * has already been enqueued, the response value of the existing request
+     * will be returned. If no existing requests match, a new request will
+     * be enqueued and executed.
+     *
+     * Equality is measured by the [AndroidManifestInfo] (excluding buildUuid)
```

Comment:
This doesn't actually exclude the buildUuid property. I tried to exclude it in f1c443b8b2793375f988632edb01437acbe0639d but that resulted in some failing tests that expected more results than would happen when it's excluded. Reverted and left for another time

Reply:
The docs are wrong on this - the buildUuid property should be part of the hashcode. There's been a bit of back and forth on the design behind this internally, but we've now settled on the following behaviour:

- By default the plugin generates a build UUID, and it is recommended that the plugin should be disabled for any unnecessary variants to improve build performance
- Advanced users can specify their own deterministic build UUID by adding a `meta-data` tag in the manifest

It's a bit of a tricky balance between the risk of overwriting existing mapping files with the same version information, versus improving performance by using cached task outputs. This approach is the best solution we've come up with so far.

In terms of actions, the comment just needs editing here.